src/nix/develop.cc

```diff
@@ -81,6 +82,10 @@ struct BuildEnvironment
             res.structuredAttrs = {json["structuredAttrs"][".attrs.json"], json["structuredAttrs"][".attrs.sh"]};
         }
 
+        if (json.contains("meta") && json["meta"].contains("min-bash-version")) {
```

Comment:
Why do we need this when we can get `NIX_ENV_MIN_BASH_VERSION` from the environment?

Reply:
We could, I just find it utterly ugly to just rely on some magic distinguished environment variables (I know we're doing that everywhere in Nix, but still)

Reply:
Hmm, as the value is already present in "variables", I agree with @edolstra in this case. Using the variable makes it clear here that this is really just about the variable. Having the `meta` in between feels round-about.

I would disagree with extracting something from the environment in this function, as it would make it impure, meaning harder to test and prone for surprising unintended functionality.